utils/compare.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package utils
+
+import (
+	"reflect"
```

Comment:
I wonder if this package is big. Have you tested if it increases binary size significantly?

We might not need the comparison function to be so generic if all we support is strings and lists of strings.

Reply:
Comparing the size with master, the unstripped difference is ~2K, where as the stripped diff is approximately 1K.

Reply:
Hehe, ok, I believe we can live with that! :stuck_out_tongue_winking_eye: